src/snowflake/connector/cursor.py

```diff
@@ -1054,9 +1056,13 @@ def __cancel_query(self, query):
             with self._lock_canceling:
                 self._connection._cancel_query(query, self._request_id)
 
-    def _log_telemetry_job_data(self, telemetry_field, value):
+    def _log_telemetry_job_data(self, telemetry_field, value=None):
         """Builds an instance of TelemetryData with the given field and logs it."""
-        obj = {"type": telemetry_field, "query_id": self._sfqid, "value": int(value)}
+        obj = {
+            "type": telemetry_field,
+            "query_id": self._sfqid,
+            "value": 1 if value is None else int(value),
```

Comment:
I like Kris' idea more of introducing a constant variable `1` for `True` over this.
If you do it this way there's no way to report `False` (which we could also declare as a constant `0`).

Reply:
int(False) is 0, int(True) is 1, these values will be recorded correctly if they are passed into the call. But when value does not matter(using default 'None'), then we just fill the value with something, and it is '1' in this case. I think defining a dummy constant is not necessary when a telemetry query on this type shall just simply ignore the 'value'.